compiler/rustc_hir_typeck/src/inline_asm.rs

```diff
@@ -93,6 +93,14 @@ impl<'a, 'tcx> InlineAsmCtxt<'a, 'tcx> {
                 }
             }
             ty::Adt(adt, args) if adt.repr().simd() => {
+                if !adt.is_struct() {
+                    self.fcx.dcx().span_delayed_bug(
+                        span,
+                        format!("repr(simd) should only be used on structs, got {}", adt.descr()),
+                    );
+                    return Err(NonAsmTypeReason::Invalid(ty));
```

Comment:
can you add `NonAsmTypeReason::Tainted(ErrorGuaranteed)` and return that here. This avoids the unnecessary inline asm error